SConstruct

```diff
@@ -414,13 +414,10 @@ if env["prereqs"]:
             client_env.Append(CPPDEFINES = ["HAVE_HISTORY"])
 
     if env["forum_user_handler"]:
-        flags = env.ParseFlags("!mysql_config --libs --cflags")
-        try: # Some versions of mysql_config add -DNDEBUG but we don't want it
-            flags["CPPDEFINES"].remove("NDEBUG")
-        except ValueError:
-            pass
+        mysql_config = check_output(["mysql_config", "--libs", "--cflags"]).replace("\n", " ").replace("-DNDEBUG", "")
+        mysql_flags = env.ParseFlags(mysql_config)
```

Comment:
What does this change gain? Instead of letting SCons run the command and parse its output, you treat the output as a string that's harder to manipulate. I don't see any semantical changes, either.

Reply:
It's parsing the output of the command as a string either way, but this way doesn't require a try/except in order to remove `-DNDEBUG`.

Reply:
I still prefer the try/except.

Reply:
Can I ask why?

Reply:
In short, I strongly prefer code that uses higher-level types instead of untyped strings. A string is not a general-purpose data container.

BTW, I'm surprised that Python's `remove()` method raises an exception if the value isn't found in the container. Ruby's equivalent method, [`Array.delete`](http://ruby-doc.org/core-2.5.0/Array.html#method-i-delete), doesn't do that.